store/sqlstore/user_store.go

```diff
@@ -1465,13 +1465,13 @@ func (us SqlUserStore) SearchNotInGroup(groupID string, term string, options *mo
 	return us.performSearch(query, term, options)
 }
 
-func generateSearchQuery(query sq.SelectBuilder, terms []string, fields []string, isPostgreSQL bool) sq.SelectBuilder {
+func generateSearchQuery(query sq.SelectBuilder, terms []string, fields []string, isPostgreSQL bool, textSearchConfig string) sq.SelectBuilder {
 	for _, term := range terms {
 		searchFields := []string{}
 		termArgs := []interface{}{}
 		for _, field := range fields {
 			if isPostgreSQL {
-				searchFields = append(searchFields, fmt.Sprintf("lower(%s) LIKE lower(?) escape '*' ", field))
+				searchFields = append(searchFields, fmt.Sprintf("to_tsvector('%[1]s', lower(%[2]s)) @@ to_tsquery('%[1]s', concat( lower(?) ,':*'))", textSearchConfig, field))
```

Comment:
Adding the `:*` prefix search seems like a diverging behavior from MySQL. Let's remove this.

Reply:
@agnivade I have some questions about the tests. What direction is best?

- mock a function e.g. `GetDefaultTextSearchConfig` to return a giver language
- create a new function like `SetDefaultTextSearchConfig` and call it whenever I want to change language during the test
- add default text search config inside the model.UserSearchOptions to be accessible from the tests

Reply:
You can manually set a different search configuration with `SET default_text_search_config`. We can run couple of tests with different configurations and then verify the results. Every time you initialize the store, it will read the search config from the DB again.

Reply:
@agnivade I've made some changes but I am maybe missing something.  Specifically
E.g. name `Jérémie` queried by the term `jere` retrieves the correct result for `default_text_search_config` equal to `spanish` but not for `french`. I do not understand why, maybe it is the expected behavior.

Reply:
> . name Jérémie queried by the term jere retrieves the correct result for default_text_search_config equal to spanish but not for french. 

Can you manually test this in the database?

> Also about the :* MySQL adds a % right after the term so I suppose MySQL also checks for prefix, right?

Ah yes that's right. So we need to remove the `%` in case of Postgres, and add the prefix search param in tsquery.

Reply:
Yes I tested manually. Some of the results are below.
Using the query `to_tsvector({language}, firstname) @@ to_tsquery({language}, concat(lower('jere'),':*'))`

Users:
Firstname: `Jérémie` , Language: `french`, query_term: `jere` results = 0
Firstname: `Jérémie` , Language: `spanish`, query_term: `jere` results = 1
Firstname: `Jerémie` , Language: `french`, query_term: `jere` results = 1
Firstname: `Jerémie` , Language: `spanish`, query_term: `jere` results = 1

Reply:
Ok, there you go then. Seems to be expected behavior.